pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -272,11 +321,17 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule, push *model.PushContex
 						CacheDuration: &durationpb.Duration{Seconds: 5 * 60},
 					},
 				}
-			} else {
+			} else if remoteJwksMode == Hybrid {
 				provider.JwksSourceSpecifier = push.JwtKeyResolver.BuildLocalJwks(jwtRule.JwksUri, jwtRule.Issuer, "")
+			} else {
+				// Log error and create fake inline JWKs
+				authnLog.Errorf("Failed to look up Envoy cluster %v. "+
+					"Please create ServiceEntry to register external JWKs server or "+
+					"set PILOT_JWT_ENABLE_REMOTE_JWKS to hybrid/istiod mode.", err)
+				provider.JwksSourceSpecifier = push.JwtKeyResolver.BuildLocalJwks("", "", model.CreateFakeJwks(jwtRule.JwksUri))
```

Comment:
How does FakeJwks differ from telling envoy to use `Cluster X` when `Cluster X` does not exist?

Reply:
Same question.

Reply:
Fake placeholder should be prohibited

Reply:
Without fake one, there is no entry provider.JwksSourceSpecifier. I added this to keep consistent with other conditions. BuildLocalJwks() will build fake one if not able to fetch the jwks. So what do we want here?

Reply:
Want to see what happens if we point it to a cluster that does not exist. I am not sure what happens, so cannot recommend the best option without knowing that.

Reply:
From my test is missing that field, requests fail authn which is expected. What type of things should be looking for making this decision?
Also if fake placeholder should be prohibited, should the BuildLocalJwks() also need to be updated?

Reply:
What is the reason to have fake jwks in the first place?

Reply:
the fake placeholder is to do the same thing -- make it fail. But its a huge hack. There is some discussion of removing https://github.com/istio/istio/issues/42202

Reply:
Without creating fake JWKs:
1. Envoy will keep giving warning about jwks_source_specifier is required as following:
```
warning	envoy config external/envoy/source/common/config/grpc_subscription_impl.cc:128	gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Error adding/updating listener(s) 0.0.0.0_8080: Proto constraint validation failed (JwtAuthenticationValidationError.Providers[origins-0]: embedded message failed validation | caused by field: "jwks_source_specifier", reason: is required): providers {
  key: "origins-0"
  value {
    issuer: "test-issuer-1@istio.io"
    payload_in_metadata: "test-issuer-1@istio.io"
  }
}
rules {
  match {
    prefix: "/"
  }
  requires {
    requires_any {
      requirements {
        provider_name: "origins-0"
      }
      requirements {
        allow_missing {
        }
      }
    }
  }
}
bypass_cors_preflight: true

	thread=16
```

2. Request with token will get following error:
```
error	envoy filter external/envoy/source/extensions/filters/http/common/jwks_fetcher.cc:51	fetch: fetch pubkey [uri = https://example.com:443/] failed: [cluster = outbound|443||example.com] is not configured	thread=30
```

Reply:
@howardjohn @hzxuzhonghu WDYT above

Reply:
Just to be clear, I meant to  remove this if part and have a LookupCluster that falls back to `model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service, port)` (I think you need to strip the `namespace` from `service`, but basically this). Then we send it, and Envoy just doesn't know about it.

So `jwks_source_specifier` isn't empty

Reply:
I see, misunderstood before. 
That makes sense, with non-existing (fake) cluster here are the findings:
1. Istiod error log
```
2023-01-18T00:21:22.440092Z	error	envoy filter external/envoy/source/extensions/filters/http/common/jwks_fetcher.cc:51	fetch: fetch pubkey [uri = https://example.com:443/] failed: [cluster = outbound|443||example.com] is not configured	thread=25
```

2. when request with token, envoy error log
```
2023-01-18T00:21:22.440092Z	error	envoy filter external/envoy/source/extensions/filters/http/common/jwks_fetcher.cc:51	fetch: fetch pubkey [uri = https://example.com:443/] failed: [cluster = outbound|443||example.com] is not configured	thread=25
```

Reply:
I tihnk you have the wrong istiod log?

Reply:
ye, here is the right one:
`error	authn	Failed to look up Envoy cluster could not find service example.com in Istio service registry. Please create ServiceEntry to register external JWKs server or set PILOT_JWT_ENABLE_REMOTE_JWKS to hybrid/istiod mode.`

Reply:
So aside from the spammy logs which should be at debug level, IMO, this seems better?

Reply:
These logs are error level, but I would agree.

Reply:
We can send a PR to envoy to change it maybe

Reply:
The error is not spammy IMO, it only logs when requests come and fail.

Reply:
You could be getting 100k QPS...

Reply:
This log is helpful when there is such errors. If it doesn't log this, fake remote jwks is not much different than fake local jwks from user perspective.
Not sure what would be potential impacts on Envoy side, if we concern about the amount of error logs then I would think fake local jwks is good enough.

Reply:
@kyessenov Could you take a look, provide feedback if there is any impact changing following error log to debug level.

when request with token, envoy error log
```
2023-01-18T00:21:22.440092Z	error	envoy filter external/envoy/source/extensions/filters/http/common/jwks_fetcher.cc:51	fetch: fetch pubkey [uri = https://example.com:443/] failed: [cluster = outbound|443||example.com] is not configured	thread=25
```

Reply:
Those are normal eventual consistency cases, while this one looks like it requires user attention? I think if a user is breaking 100k of traffic in prod, they need very clear message what's wrong, and this satisfies it.

Reply:
These are not always eventually consistency issues, all the same can happen. Make a VS to `destination: bogus` and you will get NC errors. Can we just return NC as the response flag for these?

Reply:
Want to be clear that in this particular case, it is ok to not log as error since Istiod is logging as error when the cluster is not found.
I want to know whether it affects anything else in Envoy, like is there a particular reason logging as error and it may bring some problem if it is changed.

Reply:
No, it's just logging. We can send upstream PR to change the log level. It's an obscure corner case, so don't think there's a good reason for what it is.

Reply:
@howardjohn is this blocker for this pr?